Network/NetworkService.swift

```diff
@@ -63,4 +72,34 @@ class NetworkService {
     func cancel() {
         task?.cancel()
     }
+    
+    
+    //MARK: Private
+    private func makeQuery(for url: URL, params: [String: AnyObject], type: QueryType) -> NSMutableURLRequest? {
```

Comment:
How is that possible? documentation clearly says `allHTTPHeaderFields` is read-only: https://developer.apple.com/reference/foundation/nsurlrequest/1418477-allhttpheaderfields

`httpMethod` too.